### PR TITLE
Add WAF routes to terraform const.py

### PR DESCRIPTION
**Pull request by Charl1996**
See route annotation PR [here](https://github.com/dimagi/commcare-hq/pull/29753).

##### ENVIRONMENTS AFFECTED
All

This PR contains changes which I did not make but were on CCHQ master when I executed `./manage.py list_waf_allow_patterns` (I pulled master before executing). I added the relevant reviewers for the other changes.

### CODE REVIEW

File: src/commcare_cloud/commands/terraform/constants.py [hunk 1]
```diff
@@ -17,6 +19,7 @@
 ^/a/([\w\.:-]+)/cloudcare/apps/report_formplayer_error
 ^/a/([\w\.:-]+)/data/export/custom/download_data_files/$
 ^/a/([\w\.:-]+)/data_dictionary/import$
+^/a/([\w\.:-]+)/dhis2/map/(\w+)/$
```
Comment (Charl1996) [inline src/commcare_cloud/commands/terraform/constants.py:22]: My own change

File: src/commcare_cloud/commands/terraform/constants.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 from __future__ import unicode_literals
 COMMCAREHQ_XML_POST_URLS_REGEX = r"""
 ^/a/([\w\.:-]+)/api/v([\d\.]+)/form/$
+^/a/([\w\.:-]+)/api/v0.6/case/$
+^/a/([\w\.:-]+)/api/v0.6/case/([\w-]+)/$
```
Comment (proteusvacuum) [inline src/commcare_cloud/commands/terraform/constants.py:10]: @esoergel do you know why these weren't included here before? Did you add these routes to the WAF manually?

File: src/commcare_cloud/commands/terraform/constants.py [hunk 3]
```diff
@@ -28,6 +31,7 @@
 ^/a/([\w\.:-]+)/receiver/submission/?$
 ^/a/([\w\.:-]+)/settings/users/commcare/fields/$
 ^/a/([\w\.:-]+)/settings/users/commcare/upload/$
+^/a/([\w\.:-]+)/settings/users/web/upload/$
```
Comment (proteusvacuum) [inline src/commcare_cloud/commands/terraform/constants.py:34]: @orangejenny same question as above

File: src/commcare_cloud/commands/terraform/constants.py (hunk 2 above)
Comment (esoergel) [reply src/commcare_cloud/commands/terraform/constants.py:10]: Ah, I did not add these to the WAF manually, but I did decorate the view with `@waf_allow('XSS_BODY')`.  Is this a build step that's supposed to be done whenever that decorator is applied?

Comment (Charl1996) [reply src/commcare_cloud/commands/terraform/constants.py:10]: @esoergel According to [this doc](https://confluence.dimagi.com/display/internal/How+traffic+gets+to+our+system%3A+Global+Accelerator%2C+ALB%2C+and+WAF) (see section _Allow-listing blocked traffic in the WAF_) this `constants.py` file has to be updated accordingly when using the decorator for a route view.

Comment (Charl1996) [reply src/commcare_cloud/commands/terraform/constants.py:10]: @esoergel  I'm curious, is your view that's being decorated working for whatever reason you decorated it for? If you did not update this file and for argument's sake your changes works fine, I'm wondering if updating this file is technically necessary then?

Comment (esoergel) [reply src/commcare_cloud/commands/terraform/constants.py:10]: I'm not sure in what circumstances the WAF would block access to an undecorated view, but that view hasn't been released, so it's only been used lightly, in testing.

File: src/commcare_cloud/commands/terraform/constants.py (hunk 1 above)
Comment (kaapstorm) [reply src/commcare_cloud/commands/terraform/constants.py:22]: I approve this change. If you would like to merge this before any open questions about other changes are resolved, you could PR just this line, and leave other changes for later.

**Review by kaapstorm — APPROVED** [review]

**Comment by Charl1996** [comment]
I extracted my changes to [this PR](https://github.com/dimagi/commcare-cloud/pull/4755). Keeping this one open though.

**Comment by proteusvacuum** [comment]
What's blocking here? I believe if those routes are decorated they should be included here, right? @stephherbers and @esoergel I think we just need you to confirm that you just forgot to add these routes after decorating your views. 
Alternatively, please remove the decorator from your views if you don't think it's necessary.

File: src/commcare_cloud/commands/terraform/constants.py (hunk 3 above)
Comment (orangejenny) [reply src/commcare_cloud/commands/terraform/constants.py:34]: Steph's offline, but I'm reasonably confident this should be here. This view was decorated with `waf_allow` in https://github.com/dimagi/commcare-hq/pull/28832/ and it's likely just an oversight that this file wasn't also updated.

**Comment by esoergel** [comment]
I do indeed want this view decorated and did not know I needed to take any steps beyond applying the decorator.  @Charl1996 if you can apply my change with your own, that would be very helpful.

**Comment by kaapstorm** [comment]
:+1: Sounds like this change is correct. Merging.

**Pull request merged**